test/Java10andUp/src/org/openj9/test/java/lang/management/ThreadMXBean/APITestJava10.java

```diff
@@ -0,0 +1,252 @@
+				if (id != myId) {
+					checkDepth(requestedDepth, maxDepth, trace, "dumpAllThreads");
+				} else {
+					Assert.assertTrue(trace.length <= maxDepth, "dumpAllThreads: Wrong stack size when maxDepth = "+maxDepth);
+				}
+			}
+			for (ThreadInfo ti: myBean.getThreadInfo(ids, maxDepth)) {
+				StackTraceElement[] trace = ti.getStackTrace();
+				long id = ti.getThreadId();
+				Integer requestedDepthTemp = expectedDepths.get(id);
+				if (null == requestedDepthTemp) {
+					/* non-test thread */
+					continue;
+				}
+				int requestedDepth = requestedDepthTemp;
+				if (id != myId) {
+					checkDepth(requestedDepth, maxDepth, trace, "getThreadInfo");
+				} else {
+					Assert.assertTrue(trace.length <= maxDepth, "getThreadInfo: Wrong stack size when maxDepth = "+maxDepth);
+				}
+			}
+		}
+		synchronized (waitObject) {
+			waitObject.notifyAll();
+		}
+		for (Thread t: myThreads) {
+			if (Thread.currentThread().equals(t)) {
+				continue;
+			}
+			logger.debug("wait for "+t.getId());
+			t.join(10000);
+		}
+	}
+
+	@Test(groups = { "level.extended" })
+	public void testDumpLimitsWithMethodHandles() {
+		final int REQUESTED_DEPTH = 5;
+		Thread t = new MethodHandleTestThread(REQUESTED_DEPTH - 1); // Subtract 1 for the run() method.
+		checkThreadInfo(REQUESTED_DEPTH, t);
+	}
+
+	@Test(groups = { "level.extended" })
+	public void testDumpLimitsWithLambdas() {
+		final int REQUESTED_DEPTH = 5;
+		Thread t = new LambdaTestThread(REQUESTED_DEPTH - 2); // Subtract 1 for the run() method and 1 for the lambda.
+		checkThreadInfo(REQUESTED_DEPTH, t);
+	}
+
+	private void checkThreadInfo(int requestedDepth, Thread t) {
+		t.start();
+		final long ids[] = {t.getId()};
+		while (t.getState() != Thread.State.WAITING) {
+			try {
+				Thread.sleep(1);
+			} catch (InterruptedException e) {
+				// ignore
+			}
+		}
+
+		for (int maxDepth = 0; maxDepth <= requestedDepth + 1; ++maxDepth) {
+
+			for (ThreadInfo ti: myBean.dumpAllThreads(false, false, maxDepth)) {
+				StackTraceElement[] trace = ti.getStackTrace();
+				long id = ti.getThreadId();
+				if (id != ids[0]) {
+					continue;
+				} else {
+					checkDepth(requestedDepth, maxDepth, trace, "dumpAllThreads");
+				}
+			}
+			ThreadInfo ti = (myBean.getThreadInfo(ids, maxDepth))[0];
+			StackTraceElement[] trace = ti.getStackTrace();
+			checkDepth(requestedDepth, maxDepth, trace, "getThreadInfo");
+		}
+	}
+
+	private void checkDepth(int requestedDepth, int maxDepth, StackTraceElement[] trace, String msg) {
+		/* the actual depth may be greater than the requested depth due to calls inside wait() */
+		if (requestedDepth >= maxDepth) {
+			Assert.assertEquals(trace.length, maxDepth, msg+": Wrong stack size when requested depth > maxDepth = "+maxDepth);
+		} else {
+			Assert.assertTrue(trace.length <= maxDepth, msg+": Stack exceeds maxDepth = "+maxDepth);
+			Assert.assertTrue(trace.length >= requestedDepth, msg+": Stack less than minimum expected");
```

Comment:
Would `==` work here instead of `>=`?

Reply:
I tried that.
The actual depth may be greater than the expected depth due to internal calls within wait().